mlflow/store/artifact/unity_catalog_models_artifact_repo.py

```diff
@@ -21,9 +28,7 @@
 
 from mlflow.store._unity_catalog.registry.utils import get_artifact_repo_from_storage_info
 
-REGISTRY_GET_SCOPED_TOKEN_ENDPOINT = (
-    "/mlflow/unity-catalog/model-versions/generate-temporary-credentials"
-)
+_METHOD_TO_INFO = extract_api_info_for_service(UcModelRegistryService, _REST_API_PATH_PREFIX)
```

Comment:
Using this instead because I find it more reliable than hardcoding the endpoint

Reply:
Hm @jerrylian-db question on this PR, how come the enum encoding fails in our case and not for others in OSS MLflow? For example [check out the code](https://github.com/mlflow/mlflow/blob/2e9c982b6f9a010c744b2e3d1a3a1c1203538cb3/mlflow/store/tracking/rest_store.py#L137-L149) for UpdateRun where we pass `RunStatus` (which is an enum) to update the status of an MLflow run

Reply:
Here's [the client logic](https://github.com/mlflow/mlflow/blob/2e9c982b6f9a010c744b2e3d1a3a1c1203538cb3/mlflow/tracking/_tracking_service/client.py#L357) that calls the store `update_run` method, I guess it just converts the value to an int explicitly?

Reply:
The output of this command
```
message_to_json(
  UpdateRun(
      run_uuid="1234",
      run_id="1234",
      status=RunStatus.from_string("RUNNING"),
      end_time=1,
      run_name="hello world",
  )
)
```
is this `'{\n  "run_uuid": "1234",\n  "status": "RUNNING",\n  "end_time": 1,\n  "run_id": "1234",\n  "run_name": "hello world"\n}'`

My guess is that the MLflow backend handles the enums differently. Here's how the `UpdateRun` handler in MLflow uses the status
```
trackingDatabase.updateRun(
  orgId = orgId,
  experimentId,
  uuid = uuid,
  status = rpc.getStatus,
  endTime = rpc.getEndTime
)
```
Where `rpc.getStatus` simple returns the `RunStatus` enum. In the uc-model-registry, we do this:
```
rpc.operation.map(x => ModelVersionOperation.fromValue(x.id))
```
Yeah, not sure what the difference is here. When I tested sending the `operation` field as a string with CURL, it didn't work.

Reply:
> Yeah, not sure what the difference is here. When I tested sending the operation field as a string with CURL, it didn't work.

What was the error out of curiosity (let's discuss that outside of the OSS github thread)? In MlflowServiceSuite, we have some integration tests that verify we can pass a `status` to UpdateRun as a string - I wonder if we could write a similar test for the UC backend to quickly debug & see what's going on

Reply:
Result of discussion: we will rename the OSS enum names.